cirq-core/cirq/experiments/readout_confusion_matrix.py

```diff
@@ -0,0 +1,310 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to compute readout confusion matrix and use it for readout error mitigation."""
+
+import functools
+from typing import Any, Dict, Union, Sequence, List, Tuple, TYPE_CHECKING, Optional, cast
+
+import numpy as np
+import scipy.optimize
+from cirq import circuits, ops, vis
+from cirq._compat import proper_repr
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class ReadoutConfusionMatrix:
+    """Store and use confusion matrices for readout error mitigation on sets of qubits.
+
+    The confusion matrix (CM) for two qubits is the following matrix:
+
+        ⎡ Pr(00o|00a) Pr(01o|00a) Pr(10o|00a) Pr(11o|00a) ⎤
+        ⎢ Pr(00o|01a) Pr(01o|01a) Pr(10o|01a) Pr(11o|01a) ⎥
+        ⎢ Pr(00o|10a) Pr(01o|10a) Pr(10o|10a) Pr(11o|10a) ⎥
+        ⎣ Pr(00o|11a) Pr(01o|11a) Pr(10o|11a) Pr(11o|11a) ⎦
+
+    where Pr(ij | pq) = Probability of observing “ij” given state “pq” was prepared.
```

Comment:
o --> "observed"
a --> "actual" 
It helps me understand the matrix a bit better, especially because the transpose of this matrix is also often called a confusion matrix.

Reply:
you should make a note of that, seeing as you've defined `Pr(ij | pq)` which is not the actual format you're using

Reply:
I do find it a bit jarring to have the extra decoration on top of the conditional-probability notation, but perhaps that's just me.

Reply:
Removed the a/o suffixes.